mantidproject/mantid
# BASISPowderReduction: optional normalization by proton charge

**Description of work.**
- [x] source to choose flux normalization
- [x] test for reviewer

**Report to:** nobody

**To test:**
Run the following three reductions with different flux normalization types and verify that the resulting spectra (`powder_Mon`, `powder_Pro`, and `powder_Dur`) are very similar up to a rescaling operation.

```
kwargs = dict(RunNumbers='74799',
    MaskFile='/SNS/BSS/shared/autoreduce/new_masks_08_12_2015/BASIS_Mask_default_diff.xml',
    VanadiumRuns='64642')
for norm_type in ('Monitor', 'Proton Charge', 'Duration'):
    BASISPowderDiffraction(FluxNormalizationType=norm_type,
        OutputWorkspace='powder_{}'.format(norm_type[0:3]),
        **kwargs)
```

We can roughly estimate the rescaling factor by looking at the monitor counts, the proton charge, and the duration:

|   | Monitor Counts (MC)  | Proton Charge (PC)  | Duration  |
|---|---|---|---|
| sample  |  1.5M | 111  | 331  |
| vanadium  | 2.3M | 625  | 1821 |

Then the spectrum obtained with Proton Charge will roughly be `(PC(vanadium) * MC(sample)) / (PC(sample) * MC(vanadium)) = 3.7` times the spectrum obtained with Monitor Counts. A similar scaling applies to the comparison of spectrum obtained with Duration versus the spectrum obtained with Monitor Counts.

Note: values in the previous table were obtained as follows:
- Monitor Counts: loading monitor workspace and applying `Integration` algorithm
- Proton Charge: loading the events file and then use `getRun().getProtonCharge()`
- Duration: loading the events file and then use `getRun().getProperty('Duration').value`

Fixes #25429



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
